### PR TITLE
Ja/student table

### PR DESCRIPTION
**Pull request by JavedJutt**
fetching data from API using redux and redux saga and displayed on frontend. created UI component to display data then container for that component to dispatch actions and pass data as props. actions listen by saga and api call make and store data in a variable and then pass that data as parameter to reducer's action which is modifying our store.

### CODE REVIEW

File: src/components/StudentTable.tsx [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+} from "../state/ducks/student/types";
+import moment from "moment";
+type AllProps = IDispatchToProps & IStudentState;
+function StudentTable({ fetchStudents, data }: AllProps) {
+  useEffect(() => {
+    fetchStudents();
+  }, [fetchStudents]);
+
+  const specificDate = (time: string) =>
+    moment(time).format("MMM D, YYYY [at] hh.mm A");
```
Comment (ahmadfaraz629) [inline src/components/StudentTable.tsx]: create a file helper.ts
move this function to that file and import it here.

Reason this funciton is utility function not related to react it self and can be called from another components.

File: src/components/StudentTable.tsx [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+
+  const colors: Record<string, string> = {
+    "A+": "#686563",
+    "A-": "#686563",
+    "B+": "#FFF7F5",
+    "B-": "#FFF7F5",
+    F: "#FF6897",
+  };
```
Comment (ahmadfaraz629) [inline src/components/StudentTable.tsx]: create a file named data.ts
move this function to that file and import it here.

Same reason

File: src/components/StudentTable.tsx [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+        <Button
+          variant="outlined"
+          sx={{ color: "#343744", mr: "100px", my: "20px" }}
+        >
+          + Add Data
+        </Button>
+      </Grid>
+      {/* <Button  sx={{ display: "flex", align: "right" }}>
+        Outlined
+      </Button> */}
```
Comment (ahmadfaraz629) [inline src/components/StudentTable.tsx]: Remove commented code.

File: src/state/ducks/student/reducer.ts [hunk 4]
```diff
@@ -1,10 +1,44 @@
...
+        marks: 40,
+        subject: "English",
+        grade: "F",
+        time: "2022-07-18T08:07:28.3Z",
+      },
+    ],
+  },
+
   reducers: {
+    FETCH_STUDENTS_SUCCESS: (
```
Comment (ahmadfaraz629) [inline src/state/ducks/student/reducer.ts]: **FETCH_STUDENTS_SUCCESS** fells like a const variable rathar then a function nam.
change this name to **fetchStudentsSuccess** and make ajustment according everywhere.

File: src/state/ducks/student/reducer.ts [hunk 5]
```diff
@@ -1,10 +1,44 @@
...
+      state,
+      { payload }: { payload: IStudentRaw[] | any }
+    ) => {
+      state.data = state.data.concat(payload);
```
Comment (ahmadfaraz629) [inline src/state/ducks/student/reducer.ts]: Always prefer es6

can we rewritten as 
state.data = [ ...state.data, ...payload ]

**Review by ahmadfaraz629 — COMMENTED** [review]
Good work overall 👍

File: src/components/StudentTable.tsx (hunk 1 above)
Comment (JavedJutt) [reply src/components/StudentTable.tsx]: in which folder should i create these files so that my architecture will remain universal

**Review by ahmadfaraz629 — COMMENTED** [review]
LGTM 👍

**Pull request merged**